src/libcore/python/object.cpp

```diff
@@ -30,7 +31,15 @@ class PyTraversalCallback : public TraversalCallback {
 };
 
 MTS_PY_EXPORT(Object) {
-    py::class_<Class>(m, "Class", D(Class));
+    py::class_<Class, std::unique_ptr<Class, py::nodelete>>(m, "Class", D(Class))
```

Comment:
I don't think `std::unique_ptr<Class, py::nodelete>` is needed if we use a `py::return_value_policy` of `reference` consistently everywhere when returning classes